prover/crates/bin/prover_cli/tests/status/batch.rs

```diff
@@ -0,0 +1,1524 @@
+        )
+        .await;
+
+    Command::cargo_bin("prover_cli")
+        .unwrap()
+        .arg("config")
+        .arg(connection_pool.database_url().expose_str())
+        .assert()
+        .success();
+}
+
+#[test]
+#[doc = "prover_cli status"]
+fn pli_status_empty_fails() {
+    Command::cargo_bin("prover_cli")
+        .unwrap()
+        .arg("status")
+        .assert()
+        .failure();
+}
+
+#[test]
+#[doc = "prover_cli status --help"]
+fn pli_status_help_succeeds() {
+    Command::cargo_bin("prover_cli")
+        .unwrap()
+        .arg("status")
+        .arg("help")
+        .assert()
+        .success();
+}
+
+#[test]
+#[doc = "prover_cli status batch"]
+fn pli_status_batch_empty_fails() {
+    Command::cargo_bin("prover_cli")
+        .unwrap()
+        .arg("status")
+        .arg("batch")
+        .assert()
+        .failure();
+}
+
+#[test]
+#[doc = "prover_cli status batch --help"]
+fn pli_status_batch_help_succeeds() {
+    Command::cargo_bin("prover_cli")
+        .unwrap()
+        .arg("status")
+        .arg("batch")
+        .arg("--help")
+        .assert()
+        .success();
+}
+
+#[test]
+#[doc = "prover_cli status batch -n 10000"]
+fn pli_status_of_non_existing_batch_succeeds() {
+    Command::cargo_bin("prover_cli")
+        .unwrap()
+        .arg("status")
+        .arg("batch")
+        .args(["-n", "10000"])
+        .assert()
+        .success()
+        .stdout(NON_EXISTING_BATCH_STATUS_STDOUT);
+}
+
+#[test]
+#[doc = "prover_cli status batch -n 10000 10001"]
+fn pli_status_of_multiple_non_existing_batch_succeeds() {
+    Command::cargo_bin("prover_cli")
+        .unwrap()
+        .arg("status")
+        .arg("batch")
+        .args(["-n", "10000", "10001"])
+        .assert()
+        .success()
+        .stdout(MULTIPLE_NON_EXISTING_BATCHES_STATUS_STDOUT);
+}
+
+fn status_batch_0_expects(expected_output: String) {
+    Command::cargo_bin("prover_cli")
+        .unwrap()
+        .arg("status")
+        .arg("batch")
+        .args(["-n", "0"])
+        .assert()
+        .success()
+        .stdout(expected_output);
+}
+
+fn status_verbose_batch_0_expects(expected_output: String) {
+    Command::cargo_bin("prover_cli")
+        .unwrap()
+        .arg("status")
+        .arg("batch")
+        .args(["-n", "0", "--verbose"])
+        .assert()
+        .success()
+        .stdout(expected_output);
+}
+
+async fn insert_prover_job(
+    status: ProverJobStatus,
+    circuit_id: BaseLayerCircuitType,
+    aggregation_round: AggregationRound,
+    batch_number: L1BatchNumber,
+    sequence_number: usize,
+    connection: &mut Connection<'_, Prover>,
+) {
+    connection
+        .fri_prover_jobs_dal()
+        .insert_prover_job(
+            batch_number,
+            circuit_id as u8,
+            0,
+            sequence_number,
+            aggregation_round,
+            "",
+            false,
+            ProtocolSemanticVersion::default(),
+        )
+        .await;
+    sqlx::query(&format!(
```

Comment:
We try to avoid having SQL queries outside of DAL methods.
You should move queries there and remove the dependency on `sqlx`.